components/HearingsScheduled/HearingsScheduled.tsx

```diff
@@ -0,0 +1,255 @@
+                {truncateEntry(name)}
+              </a>
+            ) : (
+              <a href={`${sessionBaseURL}${id}`}>{truncateEntry(name)}</a>
+            )}
+          </p>
+          <p className={styles.location}>{truncateEntry(location)}</p>
+        </div>
+      </div>
+    </div>
+  )
+}
+
+/** Component with interactive calendar of upcoming hearings and sessions
+ * Only events from the current month forward are displayed
+ *
+ */
+export const HearingsScheduled = () => {
+  const [monthIndex, setMonthIndex] = useState(0)
+
+  const handleSelect = (
+    selectedIndex: number,
+    e: Record<string, unknown> | null
+  ): void => {
+    setMonthIndex(selectedIndex)
+  }
+
+  /* Currently this component is expecting useUpcomingEvents to return a sorted list. 
+    The field eventList.fullDate exists as a Date type in case sorting needs to be done
+    here in the future.
+   */
+  const events = useUpcomingEvents()
+
+  const eventList: EventData[] = []
+  let latestDate = new Date()
+  const indexOffset = new Date().getMonth()
+  if (events) {
+    for (let e of events) {
+      const currentDate = new Date(e.content.EventDate)
+
+      if (currentDate > latestDate) latestDate = currentDate
+
+      const date = formatDate(e.content.EventDate)
+      if (e.type === "session") {
+        eventList.push({
+          index: currentDate.getMonth() - indexOffset,
+          type: e.type,
+          name: e.content.Name,
+          id: e.content.EventId,
+          location: e.content.LocationName,
+          fullDate: currentDate,
+          year: date.year,
+          month: date.month,
+          date: date.date,
+          day: date.day,
+          time: date.time
+        })
+      } else if (e.type === "hearing") {
+        eventList.push({
+          index: currentDate.getMonth() - indexOffset,
+          type: e.type,
+          name: e.content.Name,
+          id: e.content.EventId,
+          location: e.content.Location.LocationName,
+          fullDate: currentDate,
+          year: date.year,
+          month: date.month,
+          date: date.date,
+          day: date.day,
+          time: date.time
+        })
+      }
+    }
+  }
+
+  /* Create list of months to cycle through on calendar */
+  let currentDate = new Date()
+  const monthsList = []
+  while (currentDate.getMonth() <= latestDate.getMonth()) {
+    monthsList.push(
+      `${numberToFullMonth(currentDate.getMonth())} ${currentDate
+        .getFullYear()
+        .toString()}`
+    )
+    currentDate.setMonth(currentDate.getMonth() + 1)
+  }
+
+  /* List of events in current month displayed in carousel */
+  const thisMonthsEvents = eventList.filter(e => {
+    return e.index === monthIndex
+  })
+
+  return (
+    <Container fluid>
+      <Row className="mt-5 align-content-center">
+        <Col>
+          <h1 className={`${styles.heading}`}>Hearings Scheduled</h1>
+        </Col>
+      </Row>
+      <Row className="">
+        <Col sm={5}>
```

Comment:
We can make this collapse to a single column with the image last on mobile by adjusting the breakpoints like so:

```
<Col xs={{ order: 1 }} md={{ span: 5, order: 0 }}> // image
// ...
<Col md={7}> // carousel
```

Reply:
Done. The order is fixed. I'm going to fine-tune the spacing a bit this evening so all screen widths look good